flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcSystem.java

```diff
@@ -86,8 +95,28 @@ static RpcSystem load() {
      * @return loaded RpcSystem
      */
     static RpcSystem load(Configuration config) {
-        final ClassLoader classLoader = RpcSystem.class.getClassLoader();
-        return ServiceLoader.load(RpcSystem.class, classLoader).iterator().next();
+        try {
+            final ClassLoader classLoader = RpcSystem.class.getClassLoader();
+
+            final String tmpDirectory = ConfigurationUtils.parseTempDirectories(config)[0];
+            final Path tempFile = Files.createFile(Paths.get(tmpDirectory, "flink-rpc-akka.jar"));
+            IOUtils.copyBytes(
+                    classLoader.getResourceAsStream("flink-rpc-akka.jar"),
+                    Files.newOutputStream(tempFile));
+
+            final PluginLoader pluginLoader =
+                    PluginLoader.create(
+                            new PluginDescriptor(
+                                    "flink-rpc-akka",
+                                    new URL[] {tempFile.toUri().toURL()},
+                                    new String[0]),
+                            classLoader,
+                            CoreOptions.getPluginParentFirstLoaderPatterns(config));
```

Comment:
Shouldn't we strip down the patterns to the bare minimum? I fear that some user change has a big impact.

Reply:
It is at a minimum right now (logging + Flink). Ideally, logging is loaded parent-first, flink child-first, with nothing else ever being considered.

Reply:
But just to re-iterate, could a user overwriting the core option may cause major havoc? It feels like, we should just pass some constant here that we maintain outside from user access.

Reply:
oh yes a user messing with it breaks things big time. Since we are dealing here with a mechanism that is not intended to be extended by the user I'm +1 on using a constant.